numpy/lib/arraysetops.py

```diff
@@ -254,42 +316,94 @@ def reshape_uniq(uniq):
         uniq = np.swapaxes(uniq, 0, axis)
         return uniq
 
+    consolidated_base = np.array_view_root(consolidated)
+    orig_base = np.array_view_root(orig_ar)
+    if not indirect_sort and orig_base is not consolidated_base:
+        # We're operating on a copy, so we might as well
+        # do an in-place sort on that copy
+        sort_inplace = True
+
+    # The mask is sorted iff the input ends up sorted
+    mask_is_sorted = assume_sorted or (sort_inplace and orig_base is consolidated_base)
```

Comment:
Why is `orig_base is consolidated_base` needed here?

Reply:
`numpy/lib/tests/test_arraysetops.py:TestUnique.test_unique_axis` trips without it.

That test performs an in-place sort on a temporary copy (ie: sort_inplace was forced to True). So, sort_inplace doesn't imply mask_is_sorted if the sort is done on a copy. Only when the sort is done on the input array (ie: when the input array ends up sorted) mask_is_sorted.

I tried to clarify that on the comment above.